linux/libcmt/examples/rollup-driver.c

```diff
@@ -0,0 +1,45 @@
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+#include"rollup-driver.h"
+
+int main()
+{
+	cmt_rollup_driver_t R[1];
+	uint64_t n, rxmax, txmax;
+
+	if (cmt_rollup_driver_init(R) != 0)
+		return EXIT_FAILURE;
+
+	const void *rx = cmt_rollup_driver_get_rx(R, &rxmax);
+	void       *tx = cmt_rollup_driver_get_tx(R, &txmax);
+
+	uint8_t outputs_root_hash[32] = {0};
+
+	for (;;) {
+		int type = cmt_rollup_driver_accept_and_wait_next_input(R, &n);
+
+		switch (type) {
+		case CMT_ROLLUP_ADVANCE_STATE:
+			memcpy(tx, rx, n); /* echo */
+			cmt_rollup_driver_write_output(R, n);
```

Comment:
All official examples deserve proper error handling for all functions, to no influence people not caring about errors. I noticed this example is in the documentation, so changes here should reflect the documentation.

It is possible to generate warnings about unused error codes by adding something like

```c
#define CMT_API __attribute__((__warn_unused_result__))

CMT_API int cmt_rollup_driver_write_output(struct cmt_rollup_driver *me, uint64_t n);
```

just temporality, so you can make sure all the code base is considering the errors.